### PR TITLE
add_password unit test cases added

### LINKED ISSUE #11: Create unit-tests for `add_password` operation in `PasswordStore`
Add tests for the `add_password` operation in the `PasswordStore`

`add_password`: https://github.com/SonuBardai/lockbox/blob/main/src/store/mod.rs#L69-L82
References to other tests: https://github.com/SonuBardai/lockbox/blob/main/src/store/mod.rs#L116-L293

### PR DESCRIPTION
**Pull request by postmeback**
Fixes #11 

Please review the code and let me know, if any changes required

### CODE REVIEW

**Comment by postmeback** [comment]
I did `cargo fmt`. Can you have a look ?

File: src/store/mod.rs [hunk 1]
```diff
@@ -290,4 +295,60 @@ mod tests {
...
+        let temp_file_name = temp_file.path().to_str().unwrap();
+        let mut store =
+            PasswordStore::new(temp_file_name.to_string(), TEST_MASTER_PASSWORD.to_string())
+                .unwrap();
+        // Create a sample password to add
+        let password = PasswordEntry::new(
+            "service1".to_string(),
+            Some("user1".to_string()),
+            "password1".to_string(),
+        );
```
Comment (SonuBardai) [inline src/store/mod.rs:326]: I think we can make this test parametrized using [rstest](https://crates.io/crates/rstest). For reference you can check out the tests for [`test_find_password`](https://github.com/SonuBardai/lockbox/blob/main/src/store/mod.rs#L208-L293), [`test_load_after_store_passwords`](https://github.com/SonuBardai/lockbox/blob/main/src/store/mod.rs#L152-L206). The benefit is that we can have multiple test cases, test various edge cases like passing empty password, empty username, empty service, etc. and have them run as individual tests, instead of having this one singular test case running for the functionality.

**Review by SonuBardai — CHANGES_REQUESTED** [review]
The test needs to load passwords (ie. read the newly created store, decrypt it's contents), and then add a new password to it. This is the workflow of the password store. Apologies for not having any documentation on this, creating a new issue out of this for documenting how the password store works. For now we have a [README](https://github.com/SonuBardai/lockbox/blob/main/src/store/README.md) explaining how the encryption workflow functions.

File: src/pass/mod.rs [hunk 2]
```diff
@@ -18,6 +18,10 @@ impl PasswordEntry {
     pub fn print_password(&self) {
         println!("Password: {}", self.password);
     }
+
+    pub fn password(&self) -> &str {
+        &self.password
+    }
```
[the remark this answers is not in the document]
Comment (postmeback) [reply src/pass/mod.rs:24]: Your function will be much better because in this case, we will not be exposing the password, only the boolean result. I don't want the password field to be public.

File: src/store/mod.rs [hunk 3]
```diff
@@ -108,17 +108,22 @@ impl PasswordStore {
         if let Some(passwords) = self.passwords.as_ref() {
             passwords.print_all(show_passwords);
         } else {
-            println!("No passwords found!")
+            println!("No passwords found!");
         }
     }
 }
 
+// Rest of the code for crypto, pass, and other modules
```
[the remark this answers is not in the document]
Comment (postmeback) [reply src/store/mod.rs:116]: Removed

File: src/store/mod.rs [hunk 4]
```diff
@@ -290,4 +295,60 @@ mod tests {
...
+        );
+
+        // Add the password to the store
+        store = store
+            .add_password(
+                password.service.clone(),
+                password.username.clone(),
+                password.password().to_string(), // Use the password method to access the password
+            )
+            .unwrap();
```
[the remark this answers is not in the document]
Comment (postmeback) [reply src/store/mod.rs:336]: Updated

File: src/store/mod.rs (hunk 1 above)
Comment (postmeback) [reply src/store/mod.rs:327]: `rstest` does not work with test methods that take mutable state (like &mut self or &mut variables) as arguments. Since `test_add_password` method requires modifying the PasswordStore instance, it cannot be directly parameterized using rstest in the way I described.

**Comment by SonuBardai** [comment]
@postmeback Feel free to rerequest for review when this is ready 👍

File: src/store/mod.rs (hunk 1 above)
Comment (SonuBardai) [reply src/store/mod.rs:327]: Is there a compilation error that you're seeing when you create parametrized tests? @postmeback

Comment (postmeback) [reply src/store/mod.rs:327]: No, I asked chatgpt, whether i can use `rstest` for the given scenario. The result cam out from there.

Comment (SonuBardai) [reply src/store/mod.rs:327]: Cool. I've written several parametrized tests for methods that take mutable references to self, you can check them out here if you're interested. https://github.com/SonuBardai/lockbox/blob/main/src/store/mod.rs#L129

Eitherway, if you're still having trouble with it then we can deal with it in a separate issue. For now, we don't want to make the `password` field public like we've discussed in this comment https://github.com/SonuBardai/lockbox/pull/37#discussion_r1285411317. Also, I've refactored the code quite a bit, so that's where the conflicts are coming from. I might have renamed this `add_password` function to `push`, since it's a little more readable to do something like this:
```rs
PasswordStore::new().load().push("new_password").store();
```
